src/Skeleton/index.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { FunctionComponent } from 'react'
+
+import clsx from 'clsx'
+import styles from './skeleton.module.scss'
+
+interface Props {
+  /** Sets the skeleton-pulse style */
+  type?: 'text' | 'image' | 'button' | 'button-secondary'
+  /** Sets the width of the skeleton. */
+  width?: number | string | []
+  /** Sets the height of the skeleton. */
+  height?: number | string | []
+  /** Makes the skeleton look like a circle */
+  circle?: boolean
+  [rest: string]: unknown // ...rest property
+}
+
+const Skeleton: FunctionComponent<Props> = ({ type = 'text', width, height, circle, ...rest }) => {
+  const isTextType = type === 'text'
+  const className = clsx(
+    styles.pulse,
+    styles[`${type}`],
+    isTextType && styles.transform,
+    type === 'button-secondary' && styles.button,
+    rest.className as string
```

Comment:
Don't we usually just make this className a prop on the component. Haven't seen it done this way before. It's interesting. Unless there's a specific reason to do it this way, I say we keep it consistent and just make className a prop on the component itself.

Reply:
Hey Lee, that is a good call out. I've seen it done both ways throughout the MFCL project (AccordionItem, Button, Carousel, etc) but I'd like to uphold consistency as well. So I can definitely make this a className prop on the component itself.